src/js/components/Calendar/StyledCalendar.js

```diff
@@ -180,9 +176,11 @@ const responsiveDayButtonStyle = (props) => {
 
 // when caller opts in to day hover styling, apply all state styles
 // on CalendarDay instead of active state on CalendarDayButton
+// ${(props) =>
+//   props.theme.calendar?.day?.hover?.background && 'background: inherit;'}
 const StyledDayButton = styled(Button)`
   ${(props) =>
-    props.theme.calendar?.day?.hover?.background && 'background: inherit;'}
+    props.disabledProp && disabledStyle(props.theme.button.disabled.opacity)}
```

Comment:
Since some themes (like HPE Theme) don't use opacity for disabled button but rather disabled color, we should make sure to apply the entirety of disabled theming here (given we aren't passing `disabled` through to Button directly:

I added bounds to calendar range story:

<img width="1054" alt="Screenshot 2025-05-06 at 4 13 00 PM" src="https://github.com/user-attachments/assets/3f18785f-7789-457d-bc45-99b2cbc2c22d" />

Reply:
Taylor and I discussed this more. The issue is that StyledCalendar is looking for disabled styles to be at the root level of `props.theme.button.disabled` for themes that have kind buttons. In the hpe theme we don't define disable styles at `props.theme.button.disabled` but instead at the `props.theme.button.disabled[kind]` level. We decided to leave the grommet code as is and change the hpe theme to have disabled styles defined at `props.theme.button.disabled`. I will put up a separate PR for this.

Reply:
PR created for grommet-theme-hpe https://github.com/grommet/grommet-theme-hpe/pull/469